mkt/api/authorization.py

```diff
@@ -177,6 +165,40 @@ def __call__(self, *a):
         return self
 
 
+class GroupPermission(GroupPermissionBase):
+
+    def __init__(self, app, action):
+        self.app = app
+        self.action = action
+
+
+class AllowReviewerReadOnly(GroupPermissionBase):
+    app = 'Apps'
+    action = 'Review'
+
+    def has_permission(self, request, view):
+        return (request.method in SAFE_METHODS and
+                super(AllowReviewerReadOnly, self).has_permission(request,
+                                                                  view))
+
+
+class AllowStaffReadOnlyIfReviewed(GroupPermissionBase):
+    """
+    Allow users in the Staff group to view apps that have any published status.
+    """
+    app = 'Apps'
+    action = '%'
+
+    def has_permission(self, request, view):
+        return (request.method in SAFE_METHODS and
+                super(AllowStaffReadOnlyIfReviewed,
+                      self).has_permission(request, view))
+
+    def has_object_permission(self, request, view, object):
+        return (self.has_permission(request, view) and
+                object.status in mkt.REVIEWED_STATUSES)
```

Comment:
This PR updates the app detail api view to also use this so yes, it would be 403 if rejected. Which is a change to previously allowing reviewers always.